.github/workflows/build.yml

```diff
@@ -131,11 +131,111 @@ jobs:
       - name: Run make for java
         run: make -C lib/java
 
+      - name: Run make install for java
+        run: make -C lib/java install
+
+      - name: Upload java libthrift artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: libthrift
+          if-no-files-found: error
+          path: ~/.m2/repository/org/apache/thrift
+
       - name: Run make check for java
         run: make -C lib/java check
 
+      - name: Run make precross for java
+        run: make -C lib/java precross
+
+      - name: Upload java precross artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: java-precross
+          if-no-files-found: error
+          path: |
+            lib/java/build/functionalTestJar/
+            lib/java/build/runclient
+            lib/java/build/runnonblockingserver
+            lib/java/build/runserver
+            lib/java/build/runservletserver
+          retention-days: 3
+
       - name: Run make for kotlin
         run: make -C lib/kotlin
 
       - name: Run make check for kotlin
         run: make -C lib/kotlin check
+
+      - name: Run make precross for kotlin
+        run: make -C lib/kotlin precross
+
+      - name: Upload kotlin precross artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: kotlin-precross
+          if-no-files-found: error
+          path: |
+            lib/kotlin/cross-test-client/build/install/TestClient/
+            lib/kotlin/cross-test-server/build/install/TestServer/
+          retention-days: 3
+
+  cross-test:
+    needs:
+      - lib-java-kotlin
+    strategy:
+      matrix:
+        test_regex:
+          - "[-](binary)"
+          - "[-](header|json)"
+          - "[-](compact|multiplexed)"
```

Comment:
These patterns seem to be working now, but I'm not quite clear on why have these run as a matrix build at all? Why not just run all of the tests?

Also, "multiplexed" and "header" patterns didn't seem to match on any tests. I'm not sure if these patterns cover all the tests, or if there's some tests that failed to match and therefore didn't run.

Reply:
i'm currently just following up the travis setup, the split is also for the more anticipated cross tests in future.

i can revert back to having just one big test runs though.

Reply:
It's fine. I was just curious what the reasoning was.